tests/TestCase/ORM/EntityTest.php

```diff
@@ -1591,6 +1591,16 @@ public function testGetSetInvalid()
             'title' => 'albert',
             'body' => 'einstein'
         ], $entity->getInvalid());
+
+        $overwrite = $entity->setInvalid([
+            'title' => 'nikola',
+            'body' => 'tesla'
+        ], true);
```

Comment:
An assertion for `$overwrite === false` with the set state would be nice to make sure the state doesn't get overwritten.